src/runtime/crdt/crdt-collection.ts

```diff
@@ -259,17 +259,18 @@ export function simplifyFastForwardOp<T>(fastForwardOp: CollectionFastForwardOp<
   }
   // Sort the add ops in increasing order by the actor's version.
   const addOps = [...fastForwardOp.added].sort(([elem1, v1], [elem2, v2]) => (v1[actor] || 0) - (v2[actor] || 0));
-  const expectedClock = {...fastForwardOp.oldClock};
+  let expectedVersion = fastForwardOp.oldClock[actor];
   for (const [elem, version] of addOps) {
-    expectedClock[actor]++;
-    if (!sameVersions(expectedClock, version)) {
+    if (++expectedVersion !== version[actor]) {
       // The add op didn't match the expected increment-by-one pattern. Can't
       // replay it properly.
       return null;
     }
   }
   // If we reach here then all added versions are incremented by one.
   // Check the final clock.
+  const expectedClock = {...fastForwardOp.oldClock};
+  expectedClock[actor] = expectedVersion;
   if (!sameVersions(expectedClock, fastForwardOp.newClock)) {
```

Comment:
I think this also just needs to check expectedClock == fastForwardOp.newClock[actor]
since we know all other actors haven't added or removed anything

Reply:
nah, I take it back, you need to check everyone.